### PR TITLE
Improve documentation for NDEFReader.

### PR DESCRIPTION
**Pull request by jpmedley**
> Issue number that this PR fixes (if any). For example: 'Fixes #987654321'

NA

> What was wrong/why is this fix needed? (quick summary only)

This PR adds additional information related to the interface.

> Anything else that could help us review it

NA

### CODE REVIEW

**Comment by jpmedley** [comment]
@Elchi3 Please don't review this. I just realized that it's missing something.

File: files/en-us/web/api/ndefreader/index.html [hunk 1]
```diff
@@ -9,24 +9,24 @@
...
-<p><em>In addition to the properties listed below, <code>NDEFReader</code> inherits properties from its parent interface, {{DOMxRef("EventTarget")}}.</em></p>
+<p><em>Inherits properties from its parent, {{DOMxRef("EventTarget")}}.</em></p>
 
 <dl>
  <dt>{{DOMxRef("NDEFReader.onreading")}} {{Experimental_Inline}}</dt>
- <dd>An event handler for <code>reading</code> event, that notifies about availability of a new reading.</dd>
+ <dd>An {{domxref("EventHandler")}} called when the <code>reading</code> event is raised.</dd>
  <dt>{{DOMxRef("NDEFReader.onreadingerror")}} {{Experimental_Inline}}</dt>
- <dd>An event handler for the <code>readingerror</code> event, which is fired when an error occurs during reading.</dd>
+ <dd>An {{domxref("EventHandler")}} called when when the <code>readingerror</code> event is raised. This occurs when a tag is in proxmimity of a reading device, but cannot be read.</dd>
```
Comment (sideshowbarker) [inline files/en-us/web/api/ndefreader/index.html:29]:
```suggestion
 <dd>An {{domxref("EventHandler")}} called when when the <code>readingerror</code> event is raised. This occurs when a tag is in proximity of a reading device, but cannot be read.</dd>
```

File: files/en-us/web/api/ndefreader/index.html [hunk 2]
```diff
@@ -35,11 +35,46 @@ <h2 id="Methods">Methods</h2>
...
  <dt>{{DOMxRef("NDEFReader.write", "NDEFReader.write()")}} {{Experimental_Inline}}</dt>
- <dd>Called to write NDEF message to a tag (after ensuring hardware and UA compatibility and obtaining permission from the user) or get an error explaining why feature is not available.</dd>
+ <dd>Attempts to write an NDEF message to a tag and returns a {{jsxref("Promise")}} that either resolves when a message has been written to the tag or rejects if a hardware or permission error is encountered. This method triggers a permission prompt if the "nfc" permission has not been previously granted.</dd>
 </dl>
 
+<h2 id="Examples">Examples</h2>
+
+<h3>Handling initial reads while writing</h3>
+
+<p>The example below shows how to coordinate between a common reading handler and one used specifically for a single write. In order to write, a tag needs to be found and read. This gives you the ability to check whether it is actually a tag that you want to write to. That's why it's recommended that you call write() from a reading event.</p>
```
Comment (sideshowbarker) [inline files/en-us/web/api/ndefreader/index.html:47]:
```suggestion
<p>The example below shows how to coordinate between a common reading handler and one used specifically for a single write. In order to write, a tag needs to be found and read. This gives you the ability to check whether it is actually a tag that you want to write to. That's why it's recommended that you call <code>write()</code> from a reading event.</p>
```

File: files/en-us/web/api/ndefreader/ndefreader/index.html [hunk 3]
```diff
@@ -10,14 +10,14 @@
 <p>{{draft}}{{securecontext_header}}{{APIRef()}}</p>
 
 <p class="summary"><span class="seoSummary">The <strong><code>NDEFReader()</code></strong>
-    constructor of the <a href="/en-US/docs/Web/API/WebNFC_API">Web NFC API</a> returns a
-    newly constructed {{DOMxRef("NDEFReader")}} object used to read NDEF messages from
+    constructor of the {{domxref("NDEFReader")}} interface returns a
+    new <code>NDEFReader</code> object, which is used to read NDEF messages from
     compatiable NFC devices, e.g. NDEF tags, within the reader's magnetic induction
```
Comment (sideshowbarker) [inline files/en-us/web/api/ndefreader/ndefreader/index.html:15]:
```suggestion
    compatible NFC devices, e.g. NDEF tags, within the reader's magnetic induction
```

File: files/en-us/web/api/ndefreader/onreading/index.html [hunk 4]
```diff
@@ -9,7 +9,26 @@
 ---
 <p>{{Draft}}{{securecontext_header}}{{SeeCompatTable}}{{APIRef()}}</p>
 
-<p class="summary"><span class="seoSummary">The <code>onreading</code> property of {{DOMxRef("NDEFReader")}} interface of the <a href="/en-US/docs/Web/API/WebNFC_API">Web NFC API</a> is called whenever a new reading is available from compatible NFC devices, e.g. NFC tags supporting NDEF, when these devices are within the reader's magnetic induction field.</span></p>
+<p class="summary">The <code>onreading</code> property of {{DOMxRef("NDEFReader")}} interface is called whenever a new reading is available from compatible NFC devices, e.g. NFC tags supporting NDEF, when these devices are within the reader's magnetic induction field.</p>
```
Comment (sideshowbarker) [inline files/en-us/web/api/ndefreader/onreading/index.html:12]:
```suggestion
<p class="summary">The <code>onreading</code> property of the {{DOMxRef("NDEFReader")}} interface is called whenever a new reading is available from compatible NFC devices, e.g. NFC tags supporting NDEF, when these devices are within the reader's magnetic induction field.</p>
```

File: files/en-us/web/api/ndefreader/scan/index.html [hunk 5]
```diff
@@ -9,13 +9,12 @@
 ---
 <p>{{Draft}}{{securecontext_header}}{{SeeCompatTable}}{{APIRef()}}</p>
 
-<p><span class="seoSummary">The <code>scan()</code> method of {{DOMxRef("NDEFReader")}}
-    interface reads NDEF records from compatible NFC devices, e.g., NDEF NFC tags.</span>
+<p><span class="seoSummary">The <code>scan()</code> method of {{DOMxRef("NDEFReader")}} interface activates a reading device and returns a {{jsxref("Promise")}} that either resolves when an NFC tag is read or rejects if a hardware or permission error is encountered. This method triggers a permission prompt if the "nfc" permission has not been previously granted.</span>
```
Comment (sideshowbarker) [inline files/en-us/web/api/ndefreader/scan/index.html:12]:
```suggestion
<p><span class="seoSummary">The <code>scan()</code> method of the {{DOMxRef("NDEFReader")}} interface activates a reading device and returns a {{jsxref("Promise")}} that either resolves when an NFC tag is read or rejects if a hardware or permission error is encountered. This method triggers a permission prompt if the "nfc" permission has not been previously granted.</span>
```

File: files/en-us/web/api/ndefreader/scan/index.html [hunk 6]
```diff
@@ -24,36 +23,54 @@ <h3 id="Parameters">Parameters</h3>
...
+  <dt><code>"NotSupportedError"</code></dt>
   <dd>There is no NFC adapter compatible with Web NFC, or a connection can not be
     established.</dd>
 </dl>
 
+<h2 id="Examples">Examples</h2>
+
+<h3>Handle scanning errors</h3>
+
+<p>This example shows what happens when a scan promise rejects and `readingerror` is thrown.</p>
```
Comment (sideshowbarker) [inline files/en-us/web/api/ndefreader/scan/index.html:59]:
```suggestion
<p>This example shows what happens when a scan promise rejects and <code>readingerror</code> is thrown.</p>
```

File: files/en-us/web/api/ndefreader/write/index.html [hunk 7]
```diff
@@ -9,60 +9,150 @@
 ---
 <p>{{Draft}}{{securecontext_header}}{{SeeCompatTable}}{{APIRef()}}</p>
 
-<p>The <code>write()</code> method of {{DOMxRef("NDEFReader")}} interface writes a
-  specified message to a compatible NFC tag.</p>
+<p>The <code>write()</code> method of {{DOMxRef("NDEFReader")}} interface attempts to write an NDEF message to a tag and returns a {{jsxref("Promise")}} that either resolves when a message has been written to the tag or rejects if a hardware or permission error is encountered. This method triggers a permission prompt if the "nfc" permission has not been previously granted.</p>
```
Comment (sideshowbarker) [inline files/en-us/web/api/ndefreader/write/index.html:12]:
```suggestion
<p>The <code>write()</code> method of the {{DOMxRef("NDEFReader")}} interface attempts to write an NDEF message to a tag and returns a {{jsxref("Promise")}} that either resolves when a message has been written to the tag or rejects if a hardware or permission error is encountered. This method triggers a permission prompt if the "nfc" permission has not been previously granted.</p>
```

File: files/en-us/web/api/ndefreader/write/index.html [hunk 8]
```diff
@@ -9,60 +9,150 @@
...
+  </dd>
   <dt><code>options</code> {{optional_inline}}</dt>
   <dd>
     <ul>
-      <li><code>overwrite</code> -- {{JSxRef("Boolean")}} specifying whether or not
-        existing record should be overwritten, if such exists.</li>
-      <li><code>signal</code> -- optional {{DOMxRef("AbortSignal")}} that allows to cancel
-        this write operation.</li>
+      <li><code>overwrite</code> -- A {{JSxRef("Boolean")}} specifying whether or not
+        existing records should be overwritten, if such exists.</li>
```
Comment (sideshowbarker) [inline files/en-us/web/api/ndefreader/write/index.html:65]:
```suggestion
      <li><code>overwrite</code> -- A {{JSxRef("Boolean")}} specifying whether or not
        existing records should be overwritten, if such exist.</li>
```

**Review by sideshowbarker — APPROVED** [review]

File: files/en-us/web/api/ndefreader/write/index.html (hunk 8 above)
Comment (jpmedley) [reply files/en-us/web/api/ndefreader/write/index.html:65]: I don't think this one is right.

**Comment by jpmedley** [comment]
Please take another look.

File: files/en-us/web/api/ndefreader/write/index.html (hunk 8 above)
Comment (sideshowbarker) [reply files/en-us/web/api/ndefreader/write/index.html:65]: > I don't think this one is right.

OK

**Pull request merged**